transformer_engine/pytorch/attention.py

```diff
@@ -380,32 +560,64 @@ def forward(
             query_layer, key_layer, value_layer = [x.transpose(0,1).contiguous()
                            for x in (query_layer, key_layer, value_layer)]
 
-        batch_size, seqlen = query_layer.shape[0], query_layer.shape[1]
+        batch_size, max_seqlen_q, max_seqlen_kv = (
+            query_layer.shape[0], query_layer.shape[1], key_layer.shape[1]
+        )
 
         # [b, sq, np, hn]
         query_layer, key_layer, value_layer = [
             x.view(x.shape[0] * x.shape[1], *x.shape[2:])
             for x in [query_layer, key_layer, value_layer]
         ]
 
-        max_seqlen = seqlen
-        cu_seqlens = torch.arange(
-            0,
-            (batch_size + 1) * seqlen,
-            step=seqlen,
-            dtype=torch.int32,
-            device=query_layer.device)
+        if self.attn_mask_type != 'padding':
+            cu_seqlens_q = torch.arange(
+                0,
+                (batch_size + 1) * max_seqlen_q,
+                step=max_seqlen_q,
+                dtype=torch.int32,
+                device=query_layer.device)
+            cu_seqlens_kv = torch.arange(
+                0,
+                (batch_size + 1) * max_seqlen_kv,
+                step=max_seqlen_kv,
+                dtype=torch.int32,
+                device=key_layer.device)
+            query_layer_packed, key_layer_packed, value_layer_packed = (
```

Comment:
But ragged means something different right? Here we are basically taking the padded input and packing the valid tokens in every sample back to back for which `packed` makes sense to me.

Reply:
I initially thought `ragged` made sense because you were only packing the `valid` tokens back to back so that's what we were referring to as `ragged` on the kernels side. But I understand your thinking. I just think at some point we need to clean up the meanings of `packed` and `interleaved` a little bit. They are a little overloaded at the moment.